### PR TITLE
[ADD] New module: pos_save_client_error_fix

### PR DESCRIPTION
**Pull request by zamberjo**
The PoS displays a generic error message when an error occurs while creating users.

This module improves the error handling providing a useful message to the end user.

### CODE REVIEW

File: pos_save_client_error_fix/README.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==============
+POS Save Client Error Message FIX
+==============
+
+This module extends the functionality of POS and displays the error message returned by the server rather than "Your Internet connection is probably down.".
+
```
Comment (rafaelbn) [inline pos_save_client_error_fix/README.rst:10]: I've tested in runbot and I get the message "You Internet connection is probable down" so I don't know what to expect with this module.

Could you please explain in the "Usage" seccion how to reproduce the functionality of the module please?

**Comment by rafaelbn** [comment]
Thanks @zamberjo for this contribution. Please, rebase and after this review travis error to get all green.

In my opinion we should remove "fix" from the module's name. This module is not a fix is an improvement as I understand. The readme is not very clear but we are talking about "creating user" so we mean customers, for me a better approach of the name is 'pos_customer_save_error. Maybe here @pedrobaeza could give an opinion

**Review by rafaelbn — CHANGES_REQUESTED** [review]
I am not able to test this module because I don't understand what I must test (improve README please)

File: pos_save_client_error_fix/static/src/js/pos_save_client_error_fix.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* © 2016 Diagram Software S.L.
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+(function ($) {
+    "use strict";
+    openerp.pos_save_client_error_fix = function (instance, module) {
```
Comment (legalsylvain) [inline pos_save_client_error_fix/static/src/js/pos_save_client_error_fix.js:6]: I think that the header is unnecessary. (function($)) and jQuery at the end.

See more lighter header here : 
https://github.com/OCA/pos/blob/8.0/pos_customer_required/static/src/js/pos_customer_required.js#L1

File: pos_save_client_error_fix/static/src/js/pos_save_client_error_fix.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                if (this.uploaded_picture) {
+                    fields.image = this.uploaded_picture;
+                }
+
+                fields.id           = partner.id || false;
+                fields.country_id   = fields.country_id || false;
+                fields.ean13        = fields.ean13 ? this.pos.barcode_reader.sanitize_ean(fields.ean13) : false;
+
+                new instance.web.Model('res.partner').call('create_from_ui',[fields]).then(function(partner_id){
+                    self.saved_client_details(partner_id);
```
Comment (legalsylvain) [inline pos_save_client_error_fix/static/src/js/pos_save_client_error_fix.js:35]: Is it not more correct to call this._super(partner_id) like here : 
https://github.com/OCA/pos/blob/8.0/pos_customer_required/static/src/js/pos_customer_required.js#L22

(just a question, I'm not sure).

File: pos_save_client_error_fix/README.rst (hunk 1 above)
Comment (legalsylvain) [reply pos_save_client_error_fix/README.rst:10]: @rafaelbn, how did you tested the module ? If you loose the connection, this message is normal. The module is changing message if connection is OK, but customer creation failed, as far as I understand.

**Comment by rafaelbn** [comment]
Closing as old PR without answer in the reviews

**Pull request closed without merging**